gha: always respect the given image-tag in the helm-default action

giorio94 opened the pull request:
The blamed commit modified the clustermesh upgrade/downgrade tests to use the standard helm-default action to determine the downgrade related helm flags. While this approach worked correctly on workflow dispatch events, it turned out breaking the workflow on push events, because the helm-default action appears to internally ignore the specified image-tag on certain events. Let's fix this by moving the defaulting logic to the caller side, to give more control there and avoid confusion due to the defaulting logic. All callers are modified accordingly to preserve the previous behavior on push/schedule events.

Fixes: 7dac1a274590 ("gha: uniform downgrade settings in clustermesh upgrade/downgrade test")

giorio94 committed:
gha: always respect the given image-tag in the helm-default action

The blamed commit modified the clustermesh upgrade/downgrade tests
to use the standard helm-default action to determine the downgrade
related helm flags. While this approach worked correctly on workflow
dispatch events, it turned out breaking the workflow on push events,
because the helm-default action appears to internally ignore the
specified image-tag on certain events. Let's fix this by moving the
defaulting logic to the caller side, to give more control there and
avoid confusion due to the defaulting logic. All callers are modified
accordingly to preserve the previous behavior on push/schedule events.

Fixes: 7dac1a274590 ("gha: uniform downgrade settings in clustermesh upgrade/downgrade test")
Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

marseel commented on .github/actions/helm-default/action.yaml:
```diff
@@ -21,11 +21,7 @@ runs:
-        else
-          SHA="${{ github.sha }}"
-        fi
+        SHA="${{ inputs.image-tag }}"
```
We should probably remove `if [ -n "${SHA}" ]; then` below in this case if we expect caller to always specify correct SHA?

giorio94 replied:
Makes sense, updated by the second commit. I didn't modify that part previously because the tag was already always specified in the previous implementation (either via `image-tag` or due to defaulting), and the other commit wasn't changing anything in that respect. However, I agree that it still makes sense to simplify that logic.

giorio94 committed:
gha: always configure image-related settings in helm-default action

Currently, the helm-default action configures the image-related settings
only if the image-tag parameter is actually set to a non-empty value.
However, the image-tag parameter is marked as required, and all callers
always configure it. Hence, let's drop the extra check to keep the logic
simple and straightforward. As an extra precaution, let's configure an
invalid default, to easily catch possible misuses in the future.

Similarly, let's remove the intermediate SHA variable, in favor of
directly using the image-tag input parameter. The sha output, instead,
is currently preserved to avoid requiring further changes to the
other workflows.

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

marseel reviewed (approved):
Thanks!
